docs/src/modules/java/pages/running-locally.adoc

```diff
@@ -0,0 +1,199 @@
+= Run a service locally
+
+include::ROOT:partial$include.adoc[]
+
+Running a service locally is helpful to test and debug. The following sections provide commands for starting and stopping a single service locally.
+
+== Prerequisites
+
+In order to run your service locally, you'll need to have the following prerequisites:
+
+* Docker {minimum_docker_version} or higher
+* Access to the `gcr.io/kalix-public` container registry
+
+[TIP]
+====
+The samples provided with the Kalix SDKs all have `docker-compose` files preconfigured to run your service locally. They also contain the configuration to start a local Google Pub/Sub emulator that the Kalix proxy will connect to when you make use of topic-based Eventing in your service.
+====
+
+== Starting your service
+
+To start your service locally, run the following command from the root of your project:
+
+--
+[source, command line]
+----
+mvn kalix:runAll
+----
+--
+
+This command will start your Kalix service and a Kalix Proxy using the included `docker-compose.yml` file.
+
+If you prefer, you can also start `docker-compose` directly by running `docker-compose up` in one terminal and in another terminal start your Kalix service with:
+
+--
+[source, command line]
+----
+mvn kalix:run
+----
+--
+
+== Invoking your service
+
+After you start the service it will accept invocations on `localhost:9000`. You can use cURL to invoke your service.
+
+As an example, we will use the https://github.com/lightbend/kalix-jvm-sdk/tree/main/samples/java-spring-eventsourced-customer-registry[`customer-registry`] sample. 
+
+=== Using cURL
+
+Create a customer:
+
+[.tabset]
+Linux or macOS::
++
+--
+[source,command line]
+----
+curl localhost:9000/customer/one/create \
+  --header "Content-Type: application/json" \
+  -XPOST \
+  --data '{"email":"test@example.com","name":"Test","address":{"street":"Test street 25","city":"Test City"}}'
+----
+--
+Windows 10+::
++
+--
+[source,command line]
+----
+curl localhost:9000/customer/one/create ^
+  --header "Content-Type: application/json" ^
+  -XPOST ^
+  --data '{"email":"test@example.com","name":"Test","address":{"street":"Test street 25","city":"Test City"}}'
+----
+--
+
+Retrieve an existing customer:
+
+--
+[source,command line]
+----
+curl -XGET localhost:9000/customer/one
+----
+--
+
+== Shut down the service
+
+Use `Ctrl+C` to shutdown the service. When stopping your service, it will also shutdown the Kalix Proxy container and any other container you have defined in `docker-compose.yml`. Unless you have chosen to start `docker-compose` manually in a separate terminal, in which case you will need to stop it manually as well. 
```

Comment:
```suggestion
Use `Ctrl+C` to shut down the service. When stopping your service, it will also shutdown the Kalix Proxy container and any other container you have defined in `docker-compose.yml`. Unless you have chosen to start `docker-compose` manually in a separate terminal, in which case you will need to stop it manually as well. 
```